metaflow/plugins/aws/aws_utils.py

```diff
@@ -4,6 +4,22 @@
 from metaflow.exception import MetaflowException
 
 
+def parse_s3_full_path(s3_uri):
```

Comment:
@trhodeos is there any specific reason to create a utility function? maybe we can have this snippet live right next to L:91 in `metaflow_environment.py`. it will help in making the code much more readable (avoiding an extra jump to a different file)

Reply:
This mirrors what we do for azure and gcs. I was just following convention here, happy to pull this inline if you think it's better.